packages/hardhat-core/src/internal/hardhat-network/provider/fork/rpcToTxData.ts

```diff
@@ -14,4 +16,19 @@ export function rpcToTxData(rpcTransaction: RpcTransaction): TxData {
     s: rpcTransaction.s,
     value: rpcTransaction.value,
   };
+
+  if (rpcTransaction.type !== undefined) {
+    txData.type = rpcTransaction.type;
+  }
+  if (rpcTransaction.chainId !== undefined) {
+    txData.chainId = rpcTransaction.chainId;
+  }
+  if (rpcTransaction.accessList !== undefined) {
+    txData.accessList = rpcTransaction.accessList.map((item) => [
```

Comment:
Can't we use the other accessList format in `AccessListEIP2930TxData`? It's closer to the representation used everywhere else.

```ts
export type AccessList = AccessListItem[]
export type AccessListItem = {
  address: PrefixedHexString
  storageKeys: PrefixedHexString[]
}
```
```

Reply:
I forgot to add a comment about this. The other format uses `string` instead of `Buffer`, so I used this one to avoid adding a new `io-ts` codec for these strings. But we can do that, or (maybe better) we can just transform them here.